docs/en/ingest-management/fleet/add-fleet-server-kubernetes.asciidoc

```diff
@@ -0,0 +1,392 @@
+kubectl apply -f fleet-server-dep.yaml
+------------------------------------------------------------
++
+[IMPORTANT]
+====
+Ensure the `Deployment` and all referenced `Secrets` are created in the same `Namespace`.
+====
+
+. Check the {fleet-server} Pod logs for errors and confirm in {kib} that the {fleet-server} agent appears as `Connected` and `Healthy` in *{kib} → Fleet*.
++
+[source, shell]
+------------------------------------------------------------
+kubectl logs fleet-server-xxxxx
+------------------------------------------------------------
++
+It could take a while until the {fleet-server} fully starts. If you left the {kib} browser window opened during <<add-fleet-server-kubernetes-preparations>> it will show *Connected* in case everything goes well.
++
+Do not continue with the next steps if the {fleet-server} is not completely up and healthy and appears in Fleet UI.
+
+. Expose the {fleet-server} to {agent}s through the *Fleet Server Host*:
++
+This may include the creation of a Kubernetes `service`, an `ingress` resource, or create DNS registers for FQDNs resolution. There are multiple ways to expose applications in Kubernetes.
++
+The following shows an example to expose the {fleet-server} with a `LoadBalancer` service, which will make the {fleet-server} available for both internal and external clients. Consider creating a `ClusterIP` instead of a `LoadBalancer` service if you only need to expose the {fleet-server} to {agent}s running in the same Kubernetes cluster.
++
+[source, shell]
+------------------------------------------------------------
+kubectl expose deployment fleet-server --name fleet-svc --type LoadBalancer --port 443 --target-port 8220
+------------------------------------------------------------
++
+The previous example creates a service named `fleet-svc`, listening on port `443` and forwarding the traffic to the `fleet-server` deployment's Pods on port `8220`. The listening `--port` of the service can be customized, but the `--target-port` must remain on the default port (`8220`), because it's the port used by the {fleet-server} application.
++
+The URL(s) that the clients would use to access the {fleet-server} in this example could be:
++
+* For {agent}s within the Kubernetes cluster: `https://fleet-svc.<namespace>.svc`.
+* For external {agent}s: access through whatever FQDN added on the DNS server pointing to the public IP of the `LoadBalancer`.
++
+If your environment requires the {fleet-server} to be reachable through multiple hostnames / URLs, you can create multiple *Fleet Server Hosts* in *Fleet → Setting*, and create different policies for different groups of agents.
++
+Remember that in *Production* mode, the *hostnames* used to access the {fleet-server} must be part of the server certificate as `x.509 Subject Alternative Names`.
+
+[discrete]
+[[add-fleet-server-kubernetes-troubleshoot]]
+= Troubleshooting
+
+[discrete]
+[[add-fleet-server-kubernetes-troubleshoot-common]]
+== Common Problems
+
```

Comment:
```suggestion

The following issues may occur when {fleet-server} settings are missing or configured incorrectly:

```
I think it would be nice to have a line introducing this section, especially since we may add more items over time.